test/integration_test.py

```diff
@@ -1985,17 +1985,17 @@ def test_response_security_headers(self):
                     else:
                         response = requests.get(str(endpoint / path))
                         response.raise_for_status()
-                        expected = AzulChaliceApp.security_headers | expected_headers
-                        # FIXME: Add a CSP header with a nonce value to text/html responses
-                        #        https://github.com/DataBiosphere/azul-private/issues/6
-                        if path in ['/', '/oauth2_redirect']:
-                            del expected['Content-Security-Policy']
+                        expected = AzulChaliceApp.security_headers() | expected_headers
+                        nonce = AzulChaliceApp.validate_csp(response.headers['Content-Security-Policy'],
+                                                            has_nonce=path in ['/', '/oauth2_redirect'])
+                        expected_csp = AzulChaliceApp.content_security_policy(nonce)
```

Comment:
As I said in my previous review, have `validate_csp` return the expected CSP. The post-condition of that method is that the CSP is valid and it is therefore safe to inject it into the expectation.